pootle/apps/pootle_store/models.py

```diff
@@ -1972,7 +1971,7 @@ def update(self, overwrite=False, store=None, only_newer=False, user=None,
                                                        update_unitids, user,
                                                        revision=update_revision,
                                                        submission_type=submission_type)
-            self.file_mtime = disk_mtime
+            self.file_mtime = self.get_file_mtime() or self.file_mtime
```

Comment:
Is this not something that is better done inside `get_file_mtime()` instead of being repeated three times here?